v2/cli/init_command.go

```diff
@@ -0,0 +1,256 @@
+		"",
+	}
+
+	err := core.ValidateProjectID(fields.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = canonicalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	err = file.EnsureDirectory(".brigade")
+	if err != nil {
+		return err
+	}
+
+	// Generate js/ts buffer
+	var scriptBytes []byte
+	scriptBytes, err = execTemplate(scriptTemplate, fields)
+	if err != nil {
+		return err
+	}
+	fields.Script = string(scriptBytes)
+
+	brigadeScriptFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+	// If git repo is provided, create a separate js/ts file for script
+	var overwrite bool
+	if fields.GitCloneURL != "" {
+		scriptPath := path.Join(".brigade", brigadeScriptFileName)
+		if overwrite, err = checkFileOverwrite(scriptPath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				scriptPath,
+				scriptBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+
+		// Also generate a minimal package.json file
+		var packageBytes []byte
+		packageBytes, err = execTemplate(packageTemplate, fields)
+		if err != nil {
+			return err
+		}
+
+		packagePath := path.Join(".brigade", "package.json")
+		if overwrite, err = checkFileOverwrite(packagePath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				packagePath,
+				packageBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Generate the project.yaml file
+	var projectBytes []byte
+	projectBytes, err = execTemplate(projectTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade/project.yaml path exists.
+	// Confirm overwrite if so.
+	projectPath := path.Join(".brigade", "project.yaml")
+	if overwrite, err = checkFileOverwrite(projectPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			projectPath,
+			projectBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var notesBytes []byte
+	notesBytes, err = execTemplate(notesTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a NOTES.txt file
+	notesPath := path.Join(".brigade", "NOTES.txt")
+	if overwrite, err = checkFileOverwrite(notesPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			notesPath,
+			notesBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var secretsBytes []byte
+	secretsBytes, err = execTemplate(secretsTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a secrets.yaml file
+	secretsPath := path.Join(".brigade", "secrets.yaml")
+	if overwrite, err = checkFileOverwrite(secretsPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			secretsPath,
+			secretsBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	err = addToGitignore(
+		"./.gitignore",
+		".brigade/secrets.yaml",
+		".brigade/node_modules",
```

Comment:
I'm fuzzy on this, but I am almost positive that the trailing `/` needs to be present to ignore the entire contents of a directory (which is what we're doing here).

Even if it turns out not to be strictly required, it is at least a frequently applied convention.